packages/dev/src/main.ts

```diff
@@ -278,21 +275,122 @@ export class NetlifyDev {
       projectRoot: this.#projectRoot,
       siteID: siteID ?? '0',
     })
-    this.#runtime = runtime
+
+    this.#cleanupJobs.push(() => runtime.stop())
+
+    let serverAddress: string
+
+    // If a custom server has been provided, use it. If not, we must stand up
+    // a new one, since it's required for communication with edge functions.
+    if (typeof this.#server === 'string') {
+      serverAddress = this.#server
+    } else {
```

Comment:
If this is true, should it be conditional?
```suggestion
    // If a custom server has been provided, use it. If not, we must stand up
    // a new one, since it's required for communication with edge functions.
    if (typeof this.#server === 'string') {
      serverAddress = this.#server
    } else if (this.#features.edgeFunctions) {
```

Reply:
I didn't add the condition because I think @pieh will always need the server running for his work.

In any case, I've added the condition in https://github.com/netlify/primitives/pull/233/commits/2e788ec6b608f5cbcfe26e8af590ba5f15ea0b52 and we can revisit if/when needed.